contribs/src/main/java/com/netflix/conductor/contribs/http/GenericHttpTask.java

```diff
@@ -274,15 +278,92 @@ private void setAuthToken(Input input, Workflow workflow) throws Exception {
 		input.getHeaders().put(HttpHeaders.AUTHORIZATION, "Bearer " + response.getAccessToken());
 	}
 
-	void setReasonForIncompletion(HttpResponse response, Task task) {
-		if (response.body != null) {
-			task.setReasonForIncompletion(response.body.toString());
-		} else if (StringUtils.isNotEmpty(response.error)) {
-			task.setReasonForIncompletion(response.error);
-		} else {
-			task.setReasonForIncompletion("No response from the remote service");
+    void setReasonForIncompletion(HttpResponse response, Task task) {
+        Object failureReasonParam = task.getInputData().get(CUSTOM_FAILURE_REASON_PARAMETER_NAME);
+        if (isNull(failureReasonParam)) {
```

Comment:
StringUtils.isNullOrEmpty is better here

Reply:
`failureReasonParam ` is actually an object not a string